cocos/audio/android/jni/cddandroidAndroidJavaEngine.cpp

```diff
@@ -78,271 +69,53 @@ AndroidJavaEngine::~AndroidJavaEngine()
     {
         stopAllEffects();
     }
-    cocos2d::JniMethodInfo methodInfo;
 
-    if (!getJNIStaticMethodInfo(methodInfo, "end", "()V")) {
-        return;
-    }
-
-    methodInfo.env->CallStaticVoidMethod(methodInfo.classID, methodInfo.methodID);
-    methodInfo.env->DeleteLocalRef(methodInfo.classID);
+    JniHelper::callStaticVoidMethod(helperClassName, "end");
 }
 
 void AndroidJavaEngine::preloadBackgroundMusic(const char* filePath) {
     std::string fullPath = CocosDenshion::android::getFullPathWithoutAssetsPrefix(filePath);
-
-    // void playBackgroundMusic(String,boolean)
-    cocos2d::JniMethodInfo methodInfo;
-
-    if (!getJNIStaticMethodInfo(methodInfo, "preloadBackgroundMusic", "(Ljava/lang/String;)V")) {
-        return;
-    }
-
-    jstring stringArg = methodInfo.env->NewStringUTF(fullPath.c_str());
-    methodInfo.env->CallStaticVoidMethod(methodInfo.classID, methodInfo.methodID, stringArg);
-    methodInfo.env->DeleteLocalRef(stringArg);
-    methodInfo.env->DeleteLocalRef(methodInfo.classID);
+    JniHelper::callStaticVoidMethod(helperClassName, "preloadBackgroundMusic", filePath);
 }
 
 void AndroidJavaEngine::playBackgroundMusic(const char* filePath, bool loop) {
     std::string fullPath = CocosDenshion::android::getFullPathWithoutAssetsPrefix(filePath);
-
-    cocos2d::JniMethodInfo methodInfo;
-
-    if (!getJNIStaticMethodInfo(methodInfo, "playBackgroundMusic", "(Ljava/lang/String;Z)V")) {
-        return;
-    }
-
-    jstring stringArg = methodInfo.env->NewStringUTF(fullPath.c_str());
-    methodInfo.env->CallStaticVoidMethod(methodInfo.classID, methodInfo.methodID, stringArg, loop);
-    methodInfo.env->DeleteLocalRef(stringArg);
-    methodInfo.env->DeleteLocalRef(methodInfo.classID);
+    JniHelper::callStaticVoidMethod(helperClassName, "playBackgroundMusic", filePath, loop);
```

Comment:
@fnz It is ok. It only has problem when search path is set then fullpath is different from filepath.